File: 02-Basic/python/02-binomial_coefficient.py
def find_factorial(a):
    factorial = 1
    for i in range(1, a + 1): 
        factorial *= i  
    return factorial


def binomial_coefficient(n,r):
    if n<=r:
        print("wrong valuues")
    else:
        n_fact=find_factorial(n)
        r_fact=find_factorial(r)
        fact_nMr=find_factorial(n-r)
        bi_coefficient= n_fact/r_fact*fact_nMr
        return bi_coefficient
    
print(binomial_coefficient(6,4))

    

File: 02-Basic/JavaScript/02-binomialCoefficient.js
const findFactorial=(a)=>{
    let  factorial=1
    for (let i = 1; i <= a; i++) {
        factorial=factorial*i
        
    }
    return factorial
}

function binomialCoefficient (n,r){
    if(n<=r){
        console.error("wrong values");
        
    }else{

        let fact_n= findFactorial(n)
        let fact_r= findFactorial(r)
        let factorialnMr=findFactorial(n-r)
        let biCoefficient= fact_n/fact_r*factorialnMr
        return biCoefficient
    }

}

let bi=binomialCoefficient(5,5)
console.log(bi)

